pysimplesoap/client.py

```diff
@@ -142,16 +140,37 @@ def __init__(self, location=None, action=None, namespace=None,
             f.close()
         self.cacert = cacert
 
-        # Create HTTP wrapper
-        Http = get_Http()
-        self.http = Http(timeout=timeout, cacert=cacert, proxy=proxy, sessions=sessions)
+        # if username and password add basic authentication
         if username and password:
-            if hasattr(self.http, 'add_credentials'):
-                self.http.add_credentials(username, password)
-        if cert and key_file:
-            if hasattr(self.http, 'add_certificate'):
-                self.http.add_certificate(key=key_file, cert=cert, domain='')
+            auth_string = "{0}:{1}".format(username, password)
+            auth_header = make_headers(basic_auth=auth_string)
+            self.http_headers.update(auth_header)
+
+        # set built-in http wrapper if specified
+        if transport:
```

Comment:
Getting closer -- this is what I had in mind:

if transport instance of transportbase -- http = transport
else if transport is a string -- http = get_Http( transport)
else -- raise ValueError( ...)